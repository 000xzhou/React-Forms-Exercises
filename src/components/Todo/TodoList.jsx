function TodoList() {
  return "<BoxList />";
}

export default TodoList;
