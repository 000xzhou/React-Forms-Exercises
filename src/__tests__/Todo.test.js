import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "../components/Todo/TodoList";

it("should add new todo", () => {
  const { getByLabelText, getByText } = render(<TodoList />);
  // get inputs
  const todoInput = getByLabelText("Todo:");
  const addButton = getByText("Submit");

  // add input and submit
  fireEvent.change(todoInput, { target: { value: "100" } });
  fireEvent.click(addButton);
});
