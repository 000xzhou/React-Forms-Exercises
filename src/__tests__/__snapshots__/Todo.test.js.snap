// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="text-decoration: none;"
  />
  <button>
    Edit
  </button>
  <button>
    Mark as completed
  </button>
  <button>
    x
  </button>
</DocumentFragment>
`;
